index.d.ts

```diff
@@ -103,20 +103,19 @@ declare namespace megajs {
     }
     export class MutableFile extends File {
         storage: Storage;
-        static packAttributes(attributes: Object): Buffer;
+        static packAttributes(attributes: JSON): Buffer;
         constructor(opts: FileOpts, storage: Storage);
         mkdir(opts: mkdirOpts | string, cb?: (error: err, file: Nullable<MutableFile>) => void): void;
```

Comment:
Some methods return this, this one does not. Would not be better to make every method return void? Because the nature of the library (all methods have callbacks) method chaining is never used.

Reply:
imo mkdir should return `this` so you can do something like 
```js
 Storage.root.mkdir("myFolder").upload("hello.txt", "Hello world!");
```
instead of making a directory and the finding the directory in order to upload a file

Reply:
That's the issue with returning `this` with the way the library works now: your first example may look like you are creating a folder then uploading a file to it, but the return value is the root folder, not the created folder, so the file will be uploaded to the root folder.

Is not possible to make the mkdir return the created folder because it's a async operation. The best way to handle it is moving to promises so you can do this:

```js
const folder = await storage.mkdir('fd')
// Call folder.upload twice
```

That would require moving all functions to promises or [to this hybrid pattern](https://stackoverflow.com/a/36838115).